misk/src/test/kotlin/misk/web/marshal/UnitResponseTest.kt

```diff
@@ -0,0 +1,46 @@
+package misk.web.marshal
+
+import jakarta.inject.Inject
+import misk.MiskTestingServiceModule
+import misk.inject.KAbstractModule
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import misk.web.Get
+import misk.web.Response
+import misk.web.WebActionModule
+import misk.web.WebServerTestingModule
+import misk.web.WebTestClient
+import misk.web.actions.WebAction
+import okhttp3.ResponseBody
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+@MiskTest(startService = true)
+internal class UnitResponseTest {
+  @MiskTestModule
+  val module = TestModule()
+
+  @Inject private lateinit var webTestClient: WebTestClient
+
+  @Test
+  fun returnUnitResponseBody() {
+    with(webTestClient.get("/response/as-unit-response-body")) {
+      assertThat(response.code).isEqualTo(200)
+      assertThat(response.headers.toMap()).doesNotContainKey("Content-Type")
+      assertThat(response.body!!.bytes()).isEmpty()
+    }
+  }
+
+  class ReturnAsUnitResponseBody @Inject constructor() : WebAction {
+    @Get("/response/as-unit-response-body")
+    fun call() = Response(Unit)
```

Comment:
So is this code change mostly allowing to return `Unit` in the Response? Or is it to handle a missing request/response content type annotation? or Both?

Reply:
Added tests for those two cases. Not sure how useful they are considering the `Unit` response is IMO a strictly better version, but it certainly doesn't hurt to prove that my claims above are actually true :)